### PR TITLE
Improve battery monitoring

### PR DESCRIPTION
**Pull request by shellixyz**
This PR includes the increased battery voltage resolution changes (#2668)

Changes summary:
- Calculates and exports power draw in 0.01W unit.
- Calculates and exports mWh drawn.
- New settings for battery capacity and capacity based thresholds with chosable unit (mAh or mWh): `battery_capacity`, `battery_capacity_unit`, `battery_warning_capacity`, `battery_critical_capacity`.
- Battery settings are available through OSD menu.
- Battery capacity based beeper warnings.
- Behaves exactly as before if battery capacity settings are 0 (by default).
- Calculates and exports remaining battery capacity.
- New improved battery remaining percentage calculation based on battery capacity if available.
- New OSD elements: mWh drawn, remaining battery capacity, remaining battery percentage based on capacity if available, Wh/Km calculated efficiency
- Improved OSD elements: battery gauge based on battery capacity if available, more resolution for the power indicator element.
- Battery energy drawn in OSD stats can be displayed in mAh or Wh (`osd_stats_energy_unit` setting).
- Simplified battery task.
- Battery capacity is only used when the plugged battery was full the moment it was plugged in otherwise the battery voltage is used. Remaining battery capacity element on OSD will display `NF` if the battery was not full when plugged in. It displays `NA` if battery capacity has not been specified (battery_capacity = 0).
- New/updated MSPv2 messages to read/set the new battery settings.

I tested these changes the best I could. I could not test the Wh/Km efficiency because I have no aircraft for the moment with video/OSD and it needs to get GPS speed.

OSD fonts update with new icons: https://github.com/iNavFlight/inav-configurator/pull/333

### CODE REVIEW

**Comment by shellixyz** [comment]
I also merged #2684 in so it is not needed if you merge this.

**Comment by shellixyz** [comment]
I will do some cleaning in the batteryConfig structure. I was thinking grouping the capacity/voltage items. And also moving the multiwiiCurrentMeterOutput item where it belongs. Where should it be best placed ? telemetryConfig maybe ? or systemConfig ?

```
// FIXME this doesn't belong in here since it's a concern of MSP, not of the battery code.
uint8_t multiwiiCurrentMeterOutput;     // if set to 1 output the amperage in milliamp steps instead of 0.01A steps via msp
```

**Comment by digitalentity** [comment]
I'm not sure we need `multiwiiCurrentMeterOutput` at all. It's a legacy thing and I believe it should go at some point.

**Comment by digitalentity** [comment]
@shellixyz looks like merging of MANUAL mode created a merge conflict here

File: src/main/telemetry/smartport.c [hunk 1]
```diff
@@ -601,12 +601,10 @@ void handleSmartPortTelemetry(void)
...
             case FSSP_DATAID_FUEL       :
-                if (feature(FEATURE_CURRENT_METER)) {
-                    if (telemetryConfig()->smartportFuelPercent && batteryConfig()->batteryCapacity > 0)
-                        smartPortSendPackage(id, calculateBatteryCapacityRemainingPercentage()); // Show remaining battery % if smartport_fuel_percent=ON and battery_capacity set
-                    else
-                        smartPortSendPackage(id, mAhDrawn); // given in mAh, unknown requested unit
-                }
+                if (telemetryConfig()->smartportFuelPercent)
+                    smartPortSendPackage(id, calculateBatteryPercentage()); // Show remaining battery % if smartport_fuel_percent=ON and battery_capacity set
+                else
```
Comment (teckel12) [inline src/main/telemetry/smartport.c]: Maybe this instead so it doesn't set a telemetry fuel sensor if a current sensor doesn't exist:

```
else if (feature(FEATURE_CURRENT_METER))
```

**Comment by shellixyz** [comment]
@digitalentity Can I remove `multiwiiCurrentMeterOutput` now ? It will probably break outdated communication libraries made to work with the old Multiwii protocol but it shouldn't be a big issue.

**Review by teckel12 — APPROVED** [review]

**Comment by digitalentity** [comment]
@shellixyz yes, please remove the `multiwiiCurrentMeterOutput`

File: src/main/fc/fc_msp.c [hunk 2]
```diff
@@ -399,6 +399,13 @@ static bool mspFcProcessOutCommand(uint16_t cmdMSP, sbuf_t *dst, mspPostProcessF
         }
         break;
 
+    case MSP2_INAV_VOLTAGE_METER_CONFIG:
```
Comment (digitalentity) [inline src/main/fc/fc_msp.c]: Do we need a special MSPv2 message for voltmeter config? Same settings are available via `MSP2_INAV_BATTERY_CONFIG`.

Comment (shellixyz) [reply src/main/fc/fc_msp.c]: Indeed we do not. I added it with the PR increasing the voltage resolution but now it is redundant I will remove it.

**Comment by shellixyz** [comment]
I tested again the best I could the last changes. Including I-bus telemetry values this time.

I don't plan on making more changes. If it is good for you it is good for me to merge.

**Comment by digitalentity** [comment]
Ack. Will review today

**Comment by shellixyz** [comment]
A couple of new mods in the end even though I said I would not. Should be good this time.

**Review by digitalentity — APPROVED** [review]

**Pull request merged**